tests/middleware/test_message_state.py

```diff
@@ -0,0 +1,79 @@
+import remoulade
+from remoulade.state.middleware import MessageState
+from remoulade.cancel import Cancel
+from remoulade.middleware import SkipMessage
+from remoulade.middleware import Middleware
+
+
+class TestMessageState:
+    """Class to test the middleware
+    MessageState
+    """
+
+    @staticmethod
+    @remoulade.actor
+    def add(x, y):
+        return x + y
+
+    @staticmethod
+    @remoulade.actor
+    def error(x, y):
+        raise Exception()
+
+    """ TODO: find alternative to change priority of rabbitmq_broker in order to
+           use it in this test"""
+    def test_pending_state_message(self, rabbitmq_state, redis_state_backend):
+        rabbitmq_state.declare_actor(TestMessageState.add)
+        rabbitmq_state.add_middleware(MessageState(backend=redis_state_backend))
+        msg = TestMessageState.add.send(5, 7)
+        assert redis_state_backend._get(msg.message_id)['state'] == "Pending"
+
+    def test_success_state_message(self, local_broker, redis_state_backend):
+        local_broker.declare_actor(TestMessageState.add)
+        local_broker.add_middleware(MessageState(backend=redis_state_backend))
+        msg = TestMessageState.add.send(5, 7).result
+        assert redis_state_backend._get(msg.message_id)['state'] == "Success"
+
+    def test_started_state_message(self, local_broker, redis_state_backend):
```

Comment:
Do not use a local broker. But a stub broker.
you can test it like this, it's not perfect but enougth:
```
    def test_started_state_message(self, stub_broker, stub_worker, state_middleware):
        
        @remoulade.actor
        def wait():
            time.sleep(0.3)

        stub_broker.declare_actor(wait)

        msg = wait.send()
        time.sleep(0.1)

        assert state_middleware.backend._get(msg.message_id)['state'] == "Started"
```
I can explain it to you if you want

Reply:
cool, thank u